breakerbp/failure_ratio.go

```diff
@@ -151,8 +192,13 @@ func (cb FailureRatioBreaker) stateChanged(name string, from gobreaker.State, to
 		nameLabel: cb.name,
 	}).Set(value)
 
-	message := fmt.Sprintf("circuit breaker %v state changed from %v to %v", name, from, to)
-	cb.logger.Log(context.Background(), message)
+	slog.InfoContext(
+		cb.logContext,
+		"circuit breaker state changed",
```

Comment:
🔕 snoolog is like LogDNA, it doesn't really display these well.  You'll just see a flood of "circuit breaker state changed" and will have to jump through hoops just to see "circuit breaker state changed from X to Y" which is what you basically want to see as a human. Can we format the log message so it has everything a human needs, and leave the kvs to be just for filtering?

(I'm not sure how much we want to care about this in infrared v0 for the debug case above, though, since we don't have a great way to save the allocs it would take to format it if it's not being printed.)

Reply:
when upstream is healthy the state would never change and this log won't appear at all. so when this log appears, even if it's changing from open to closed (recover), it means the state changed to open before so it's not exactly false alarm.

and have kv pairs of the from/to states make it much easier to search/filter.